other/limit-configmap-for-sa/limit_configmap_for_sa.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: limit-configmap-for-sa
+  annotations:
+    policies.kyverno.io/title: Limit ConfigMap for a User
+    policies.kyverno.io/category: Other
+    policies.kyverno.io/severity: medium
+    policies.kyverno.io/subject: ConfigMap
+    policies.kyverno.io/description: Limits given ConfigMap for a specific users.
+spec:
+  background: false
+  validationFailureAction: audit
+  rules:
+  - name: limit-configmap-for-sa-developer
+    match:
+      any:
+      - resources:
+          kinds:
+          - "ConfigMap"
+        subjects:
+        - kind: ServiceAccount
+          name: developer
+          namespace: kube-system
+      - resources:
+          kinds:
+          - "ConfigMap"
+        subjects:
+        - kind: ServiceAccount
+          name: another-developer
+          namespace: another-namespace
+    preconditions:
+      all:
+      - key: "{{request.object.metadata.namespace}}"
+        operator: In
+        value:
+        - "any-namespace"
+        - "another-namespace"
+      - key: "{{request.object.metadata.name}}"
+        operator: In
+        value:
+        - "any-configmap-name-good"
+        - "another-configmap-name"
+    validate:
+      message: "{{request.object.metadata.namespace}}/{{request.object.kind}}/{{request.object.metadata.name}} resource is protected. Admin or allowed users can change the resource"
+      deny:
+        conditions:
+          all:
+          - key: "{{request.operation}}"
+            operator: "In"
+            value:
+            - "UPDATE"
+            - "CREATE"
+            - "CONNECT"
```

Comment:
The `CONNECT` operation is not valid for a ConfigMap. This only applies to Pods when exec'ing.

Reply:
Without `CONNECT`, the user can edit the ConfigMap with `kubectl edit cm <resource-name>` or `kubectl apply -f configmap.yaml` if resource exists.

I don't it is intentionally done but it is needed to prevent users to edit ConfigMap. `UPDATE` doesn't cover `PATCH` requestes.

Reply:
> Without CONNECT, the user can edit the ConfigMap with kubectl edit cm <resource-name> or kubectl apply -f configmap.yaml if resource exists.

I don't believe that's correct. I actually just tried this and either of those two actions results in an `UPDATE` operation which would be blocked by the policy.

Reply:
I tried it again now it worked. I think I missed something at that time. I removed the `CONNECT` part. 👍